invenio_rdm_records/resources/serializers/ui/schema.py

```diff
@@ -203,3 +206,21 @@ class UIRecordSchema(BaseObjectSchema):
         fields.Nested(FundingSchema()),
         attribute="metadata.funding",
     )
+
+    @pre_dump
+    def add_communities_permissions_and_roles(self, obj, **kwargs):
+        """Inject current user's permission to community receiver."""
+        receiver = (
+            obj.get("expanded", {}).get("parent", {}).get("review", {}).get("receiver")
+        )
+        if receiver:
+            can_direct_publish = _community_permission_check(
```

Comment:
question: I am a bit out of the loop on this subject and this catched my attention. Is it common to check permissions on serialization?

Reply:
we do it when we want to return the permissions that the user making the request has, it is a way to expose them, and allow building UI correctly